var/spack/repos/builtin/packages/py-numba/package.py

```diff
@@ -28,9 +28,9 @@ class PyNumba(PythonPackage):
     # That's why it was chosen as an upper bound in the following depends_on
     # calls.  If newer versions maintain backwards compatibility, the calls
     # can be updated accordingly.
-    depends_on('py-llvmlite@0.33:0.34', type=('build', 'run'), when='@0.50.1')
+    depends_on('py-llvmlite@0.33.0', type=('build', 'run'), when='@0.50.1')
```

Comment:
llvmlite@0.34.0 is not compatible with numba@0.50.1: this is demonstrated by a runtime error and corroborated by the Dependencies section of https://pypi.org/project/numba/0.50.1/.

Reply:
```suggestion
    depends_on('py-llvmlite@0.33', type=('build', 'run'), when='@0.50.1')
```
I think this may be capturing `0.33:0.33.999` basically. No guarantees, though.

Reply:
With the version 0.33 instead of 0.33.0, we have:

`==> Installing py-llvmlite-0.33-6m7uhom7vbb7rvd4l3dp6lcj3xnrh6ku
==> No binary for py-llvmlite-0.33-6m7uhom7vbb7rvd4l3dp6lcj3xnrh6ku found: installing from source
==> Warning: There is no checksum on file to fetch py-llvmlite@0.33 safely.
==>   Fetch anyway? [y/N] y
==> Fetching https://pypi.io/packages/source/l/llvmlite/llvmlite-0.33.tar.gz
######################################################################## 100.0%
curl: (22) The requested URL returned error: 404 Not Found
==> Error: FetchError: All fetchers failed

/gpfs/bbp.cscs.ch/home/lguyot/Documents/NSE/catalog/spack/lib/spack/spack/package.py:1331, in do_fetch:
       1328                                 self.spec.format('{name}{@version}'), ck_msg)
       1329
       1330        self.stage.create()
  >>   1331        err_msg = None if not self.manual_download else self.download_instr
       1332        self.stage.fetch(mirror_only, err_msg=err_msg)
       1333        self._fetch_time = time.time() - start_time
       1334`

The available versions of llvmlite are:
..., 0.32.0, 0.32.1, **0.33.0rc1, 0.33.0,** 0.34.0rc1, 0.34.0, ...